pkg/northbound/gnmi/set.go

```diff
@@ -115,15 +115,15 @@ func (s *Server) Set(ctx context.Context, req *gnmi.SetRequest) (*gnmi.SetRespon
 	}
 
 	//TODO this can be parallelized with a pattern manager.go ValidateStores()
-	mgr := manager.GetManager()
 	//Checking for wrong configuration against the device models for updates
 	for target, updates := range targetUpdates {
 		err := validateChange(target, version, deviceType, updates, targetRemoves[target])
 		if err != nil {
 			return nil, err
 		}
 		delete(targetRemovesTmp, target)
-		if _, ok := mgr.DeviceStore.Cache[device.ID(target)]; !ok {
+		_, errDevice := manager.GetManager().DeviceStore.Get(device.ID(target))
+		if errDevice != nil && status.Convert(errDevice).Code() == codes.NotFound {
 			disconnectedDevices = append(disconnectedDevices, target)
 		}
```

Comment:
Maybe should handle the `err` for other codes too

Reply:
added ```else if errDevice != nil {
			//handling gRPC errors
			return nil, err
		}``` to handle any unexpected error, only know thrown by topo is NotFound (see line  149 of service.go under `onos-topo`